crates/wasmparser/src/validator/core.rs

```diff
@@ -784,7 +803,15 @@ impl Module {
             (ValType::Bot, _) => true,
             (_, ValType::Bot) => false,
```

Comment:
I think this case redundant, as I think it would be covered by a default case of the form `(ty1, ty2) => ty1 == ty2`